state/factory/factory.go

```diff
@@ -143,19 +143,8 @@ func NewFactory(cfg config.Config, opts ...Option) (Factory, error) {
 func (sf *factory) Start(ctx context.Context) error {
 	sf.mutex.Lock()
 	defer sf.mutex.Unlock()
-	var err error
-	if err = sf.lifecycle.OnStart(ctx); err != nil {
-		return err
-	}
-	// get state trie root
-	if sf.rootHash, err = sf.getRoot(trie.AccountKVNameSpace, AccountTrieRootKey); err != nil {
-		return errors.Wrap(err, "failed to get accountTrie's root hash from underlying DB")
-	}
-	if sf.accountTrie, err = trie.NewTrie(sf.dao, trie.AccountKVNameSpace, sf.rootHash); err != nil {
-		return errors.Wrap(err, "failed to generate accountTrie from config")
-	}
 
-	return sf.accountTrie.Start(ctx)
+	return sf.lifecycle.OnStart(ctx)
```

Comment:
So the order doesn't matter?

Reply:
The order doesn't matter as long as we handle the dependency properly. In our previous implementation, the root hash was stored in both sf and trie, which results in the messy of dependency.